src/anemoi/inference/decorators.py

```diff
@@ -43,17 +43,29 @@ def __init__(context, encoding=None, path=None, archive_requests=None):
         grib:
             path: out.grib
     ```
+
+    Also supports capturing all positional arguments into the main argument as a tuple:
+    ```
+    @main_argument("paths", capture_all_args=True)
+    class MultiFileOutput
+        def __init__(context, paths=None, archive_requests=None):
+            ...
+    output = MultiFileOutput(context, "out1.grib", "out2.grib")
+    ```
     """
 
-    def __init__(self, name: str):
+    def __init__(self, name: str, *, capture_all_args: bool = False):
```

Comment:
Can you explain the usefulness of this new flag? Not sure about the benefit of this extra complexity. The decorator should really only be used in combination with the config factory, but I don't see how it's useful in a config setting and in relation to the cutout input. 

Doesn't this already work without the new flag?

```
output: 
  grib: ["out1.grib", "out2.grib"]
```